Task-6 Добавить Koin

IvanPodlesnykh opened the pull request:
1. Создал класс App
2. Добавил в класс App инициализацию Koin
3. Добавил ссылку на класс App в AndroidManifest
4. Создал файлы модулей и сами модули: dataModule,interactorModule, repositoryModule, uiModule
5. В dataModule добавил инициализацию Gson и SharedPreferences

IvanPodlesnykh committed:
Task-6 Добавить Koin
1. Создал класс App
2. Добавил в класс App инициализацию Koin
3. Добавил ссылку на класс App в AndroidManifest
4. Создал файлы модулей и сами модули: dataModule,interactorModule, repositoryModule, uiModule
5. В dataModule добавил инициализацию Gson и SharedPreferences

IvanPodlesnykh committed:
Task-6 Добавить Koin
Поправил форматирование файла App

herrfreeman commented on app/src/main/java/ru/practicum/android/App.kt:
```diff
@@ -0,0 +1,22 @@
+import ru.practicum.android.diploma.di.repositoryModule
+import ru.practicum.android.diploma.di.uiModule
+
+class App : Application() {
```
Предлагаю сделать название класса приложения более информативным, например FindJobApplication

herrfreeman commented on app/src/main/java/ru/practicum/android/diploma/di/DataModule.kt:
```diff
@@ -0,0 +1,21 @@
+
+    single {
+        androidContext()
+            .getSharedPreferences(SHARED_PREFERENCES_KEY, AppCompatActivity.MODE_PRIVATE)
```
Для настроек мы вроде делали на курсе отдельный интерактор, напрямую через DI не пробрасывали, давай пока уберем, потом обсудим как настройки делать

lodrean replied:
У меня через DI настроены зависимости. Shared pref. Gson. Retrofit. Контекст. В общем можно и нужно через DI делать

IvanPodlesnykh committed:
Task-6 Добавить Koin Правки
1. Переименовал класс App в FindJobApplication
2. Убрал инициализацию SharedPreferences из dataModule
